packages/app-mobile/components/NotesBar.tsx

```diff
@@ -0,0 +1,278 @@
+import { Style } from './global-style';
+import Note from '@joplin/lib/models/Note';
+import NotesBarListItem from './NotesBarListItem';
+import Folder from '@joplin/lib/models/Folder';
+
+interface Props {
+    themeId: string;
+	items: any[];
+	todoCheckbox_change: (checked: boolean)=> void;
+	selectedFolderId: string;
+	activeFolderId: string;
+	dispatch: any;
+	selectedNoteId: string;
+	toggleNotesBar: ()=> void;
+	settings: any;
+}
+
+function NotesBarComponent(props: Props) {
+
+	const [notes, setNotes] = React.useState<any[]>(props.items);
+	const [query, setQuery] = React.useState<string>('');
+
+	const themeId = props.themeId;
+	const theme = themeStyle(themeId);
+
+	function styles() {
+		let styles: Style = {
+			container: {
+				flex: 1,
+				width: '100%',
+				backgroundColor: theme.backgroundColor3,
+			},
+			horizontalFlex: {
+				flexDirection: 'row',
+			},
+			title: {
+				alignItems: 'center',
+			},
+			titleText: {
+				fontSize: 16,
+			},
+			closeIcon: {
+				fontSize: 30,
+				paddingTop: 8,
+				paddingBottom: 8,
+				paddingRight: theme.marginRight,
+				paddingLeft: theme.marginLeft,
+			},
+			top: {
+				color: theme.color,
+			},
+			topContainer: {
+				width: '100%',
+				justifyContent: 'space-between',
+				paddingLeft: theme.marginLeft,
+			},
+			padding: {
+				paddingLeft: theme.marginLeft,
+				paddingRight: theme.marginRight,
+				paddingTop: 12,
+				paddingBottom: 12,
+			},
+			titleIcon: {
+				fontSize: 22,
+				marginRight: 4,
+			},
+			divider: {
+				backgroundColor: theme.dividerColor,
+				height: 1,
+				width: '100%',
+			},
+			nativeInput: {
+				fontSize: theme.fontSize,
+				flex: 1,
+				paddingRight: 8,
+			},
+			searchIcon: {
+				fontSize: 22,
+			},
+			searchInput: {
+				alignItems: 'center',
+				backgroundColor: theme.backgroundColor,
+				paddingLeft: 8,
+				borderRadius: 4,
+				borderWidth: 1,
+				borderColor: theme.dividerColor,
+				height: 42,
+				flex: 1,
+			},
+			button: {
+				height: 42,
+				width: 42,
+				backgroundColor: theme.color4,
+				alignItems: 'center',
+				justifyContent: 'center',
+				borderRadius: 4,
+				flex: 0.5,
+				marginLeft: 8,
+			},
+			buttonIcon: {
+				color: theme.backgroundColor,
+				fontSize: 22,
+			},
+			inputGroup: {
+				justifyContent: 'space-between',
+			},
+		};
+
+		styles = StyleSheet.create(styles);
+
+		return styles;
+	}
+
+	const titleComp = (
+		<View style={[styles().title, styles().horizontalFlex]}>
+			<Icon name='md-document'style={[styles().top, styles().titleIcon]} />
+			<Text style={[styles().top, styles().titleText]}>{_('Notes')}</Text>
+		</View>
+	);
+
+	const dividerComp = (
+		<View style={styles().divider}></View>
+	);
+
+	const closeButtonComp = (
+		<TouchableOpacity onPress={props.toggleNotesBar}>
+			<Icon name="close" style={[styles().top, styles().closeIcon]}/>
+		</TouchableOpacity>
+	);
+
+	function renderIconButton(icon: JSX.Element, onPress: ()=> Promise<void>) {
```

Comment:
Prefer arrow functions for new code (see the [style guide](https://github.com/laurent22/joplin/blob/dev/readme/coding_style.md#prefer----to-function---)).
```suggestion
	const renderIconButton = (icon: JSX.Element, onPress: ()=> Promise<void>) => {
```

Reply:
Noted thanks